test/functional/p2p_tx_privacy.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.messages import (
+    msg_wtxidrelay,
+    msg_verack,
+    msg_tx,
+)
+from test_framework.p2p import (
+    P2PInterface,
+    p2p_lock,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.wallet import MiniWallet
+
+class P2PTxSpy(P2PInterface):
+    def on_version(self, message):
+        self.send_message(msg_wtxidrelay())
+
+class TxPrivacyTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def run_test(self):
+        self.wallet = MiniWallet(self.nodes[0])
+        self.wallet.rescan_utxos()
+
+        tx_originator = self.nodes[0].add_p2p_connection(P2PInterface())
+        spy = self.nodes[0].add_p2p_connection(P2PTxSpy(), wait_for_verack=False)
+        spy.wait_for_verack()
+
+        # tx_originator sends a tx
+        tx = self.wallet.create_self_transfer()["tx"]
+        tx_originator.send_and_ping(msg_tx(tx))
+
+        with p2p_lock:
+            spy.last_message.pop("inv", None)
```

Comment:
What is the reason for including this `pop`? If bitcoind ever sent an inv to a peer before receiving a verack, shouldn't this test rather fail than silently remove it?

Reply:
No reason, removed it. Thanks!